c++/src/kj/mutex.c++

```diff
@@ -547,8 +547,16 @@ void Mutex::wait(Predicate& predicate, Maybe<Duration> timeout) {
     } else {
       DWORD error = GetLastError();
       if (error == ERROR_TIMEOUT) {
-        // Timed out. Skip predicate check.
-        return;
+        // As mentioned above, Windows sleeps are inaccurate, so we must double check.
+        KJ_IF_MAYBE(e, endTime) {
+          if (*e <= clock->now()) {
+            // Actually timed out. Skip predicate check.
+            return;
+          }
+        } else {
+          // Unexpected ERROR_TIMEOUT. Should we KJ_FAIL_WIN32() instead?
```

Comment:
Maybe the `KJ_IF_MAYBE(e, endTime)` can be `KJ_ASSERT_NONNULL(endTime)`?

Reply:
Done.